### PR TITLE
ROC-5471 Add skeleton 

### PR DESCRIPTION
**Pull request by simad**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-5471

### Change description ###

Adding skeleton for the doc assembly client. This can be used to generate a document via doc assembly api, store it in docstore and receive a link to it.

Added: 
- checkstyle.xml
- gradle build files

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      if (rejected) {
+        selection.reject('Release candidate')
+      }
+    }
+  }
+}
+
+checkstyle {
+  maxWarnings = 0
+  toolVersion = '8.3'
```
Comment (sabahirfan) [inline build.gradle]: Please update to latest version, I believe 8.19 is out now.

Comment (simad) [reply build.gradle]: 👍

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        root.appendNode('description', 'Client library for communicating with doc assembly api')
+        root.appendNode('name', 'Doc assembly client')
+        root.appendNode('url', 'https://github.com/hmcts/doc-assembly-client')
+        root.children().last() + pomConfig
+      }
+    }
+  }
+}
+
+bintray {
```
Comment (sabahirfan) [inline build.gradle]: I don't think bintray is necessary for this. We can remove this

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+apply plugin: 'checkstyle'
+apply plugin: 'java-library'
+apply plugin: 'org.springframework.boot'
+apply plugin: 'maven'
+apply plugin: 'maven-publish'
+apply plugin: 'com.jfrog.bintray'
+
+group 'uk.gov.hmcts.reform'
+version '5.0.0'
```
Comment (sabahirfan) [inline build.gradle]: Is this correct?

Comment (simad) [reply build.gradle]: It was a copy paste 🤦‍♀️

**Pull request merged**